cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -161,12 +163,14 @@ func Commands(globalParams *command.GlobalParams) []*cobra.Command {
 						PythonVersionGetFunc: python.GetPythonVersion,
 					},
 					status.NewInformationProvider(leaderelection.Provider{}),
-					status.NewInformationProvider(clusteragentMetricsStatus.Provider{}),
 					status.NewInformationProvider(admissionpkg.Provider{}),
 					status.NewInformationProvider(endpointsStatus.Provider{}),
 					status.NewInformationProvider(clusterchecks.Provider{}),
 					status.NewInformationProvider(orchestratorStatus.Provider{}),
 				),
+				fx.Provide(func(dogCl datadogclient.Component) status.InformationProvider {
+					return status.NewInformationProvider(clusteragentMetricsStatus.GetProvider(dogCl))
+				}),
```

Comment:
Following the @hush-hush comment https://github.com/DataDog/datadog-agent/pull/27244/files#r1682501420

Why don't we add a status provider to the datadogclient. 

I could understand that it might change a bit the structure of the human readable status, but I think it would be worth it as it would avoid having to deviate from the standard way if adding status output. 

My suggestion would be to create a status provider inside the `datadogclient` component. 

Depending on the client that is create we use one or the other

Here is how the code would look like https://github.com/DataDog/datadog-agent/pull/27700